starwars_api/models.py

```diff
@@ -55,27 +63,64 @@ def __repr__(self):
 class BaseQuerySet(object):
 
     def __init__(self):
-        pass
+        self.next_page = None
+        self.previous_page = None
+        self.results = []
+        self.current_record = 0
+        self.current_page = 0
 
     def __iter__(self):
-        pass
+        return self
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
-
+        while True:
+            if not self.results:
```

Comment:
still a bit rusty on `if` and `if not` - does `if not self.results:` mean that if `self.results` returns `None`? am a bit confused as to the logic here.